src/pages/resolve/components/IpfsLink2.vue

```diff
@@ -0,0 +1,27 @@
+<template>
+  <a :href="hashLink" class="hash-link" target="_blank">{{
+    hash.substring(0, this.length || 8)
+  }}</a>
+</template>
+
+<script>
+export default {
+  props: ["hash", "length", "ellipsis"],
+  computed: {
+    hashLink() {
+      return `https://api.dstor.cloud/ipfs/${this.hash}`;
+    },
+  },
+};
+</script>
+
+<style scoped>
```

Comment:
Verify your styles don't break anything outside this scope.

```suggestion
<style>
```